[CI] GenerativeIT test failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #7413 / almalinux-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/xpzw2wkyghg76)
- [elasticsearch-periodic-platform-support #7413 / rocky-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/dluzpio7wkiz4)
- [elasticsearch-pull-request #65917 / part-3](https://gradle-enterprise.elastic.co/s/lyug77tygqkue)
- [elasticsearch-pull-request #65913 / part-3](https://gradle-enterprise.elastic.co/s/66wg57ns7iub2)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT.test" -Dtests.seed=CFE3FB5D9710EFCD -Dtests.locale=ha-GH -Dtests.timezone=Asia/Brunei -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: query: from languages_lookup_non_unique_ke*,no_mapping_sample_data* | limit 9180 | eval  `language_code` = -2657316569294869309, jtdeY = -706707837, country = 850667825, `country.keyword` = -5713135059288987047, country.keyword = false, language_code = 6745982913013275940, zod = -3119129993057472970, rQNNd = 769646083, language_code = "vvdHygIqHqV", language_name = -1851577784 | sort language_code ASC NULLS FIRST, jtdeY ASC NULLS LAST, language_name DESC NULLS FIRST| rename jtdeY as language_code | lookup join languages_lookup on language_code | rename `country` AS uGhif, language_code AS `rQNNd` | mv_expand `country.keyword`| rename language_name as message | lookup join message_types_lookup on message | rename `uGhif` AS ectad | dissect message "%{zod}" | dissect message "%{xSULx} %{type}"| rename zod as message | lookup join message_types_lookup on message | mv_expand country.keyword | mv_expand `message` | grok message "%{WORD:SPJey} %{WORD:kgyEp} %{WORD:z
[truncated]
```

**Issue Reasons:**
- [main] 4 failures in test test (5.1% fail rate in 79 executions)
- [main] 2 failures in step part-3 (4.9% fail rate in 41 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (4.9% fail rate in 41 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -357,9 +357,6 @@ tests:
 - class: org.elasticsearch.search.basic.SearchWithRandomDisconnectsIT
   method: testSearchWithRandomDisconnects
   issue: https://github.com/elastic/elasticsearch/issues/122707
-- class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT
-  method: test
-  issue: https://github.com/elastic/elasticsearch/issues/126139
 - class: org.elasticsearch.snapshots.SharedClusterSnapshotRestoreIT
   method: testDeletionOfFailingToRecoverIndexShouldStopRestore
   issue: https://github.com/elastic/elasticsearch/issues/126204

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
@@ -41,6 +41,7 @@ public abstract class GenerativeRestTest extends ESRestTestCase {
         "Unbounded sort not supported yet",
         "The field names are too complex to process", // field_caps problem
         "must be \\[any type except counter types\\]", // TODO refine the generation of count()
+        "mismatched input .* expecting", // identifier generator needs to be refined, this happens when an identifier is a reserved keyword
 
         // warnings
         "Field '.*' shadowed by field at line .*",
@@ -52,10 +53,9 @@ public abstract class GenerativeRestTest extends ESRestTestCase {
         "only supports KEYWORD or TEXT values, found expression", // https://github.com/elastic/elasticsearch/issues/126017
         "token recognition error at: '``", // https://github.com/elastic/elasticsearch/issues/125870
         "Unknown column \\[.*\\]", // https://github.com/elastic/elasticsearch/issues/126026
-        "Expected \\[.*\\] but was \\[.*\\]", // https://github.com/elastic/elasticsearch/issues/126030
-        "trying to encode an unsupported data type value for TopN", // still https://github.com/elastic/elasticsearch/issues/126030 probably
-        "Block cannot be cast to", // https://github.com/elastic/elasticsearch/issues/126036
         "optimized incorrectly due to missing references", // https://github.com/elastic/elasticsearch/issues/116781
+        "No matches found for pattern", // https://github.com/elastic/elasticsearch/issues/126418
+        "JOIN left field .* is incompatible with right field", // https://github.com/elastic/elasticsearch/issues/126419
         "The incoming YAML document exceeds the limit:" // still to investigate, but it seems to be specific to the test framework
     );
 

